bin/check-vault-status.rb

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env ruby
+#
+# check-vault-status
+#
+# DESCRIPTION:
+#   This plugins checks if vault is up & reachable. It then checks
+#   if the vault is unsealed
+#
+# OUTPUT:
+#   plain text
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#   gem: vault
+#
+require 'sensu-plugin/check/cli'
+require 'vault'
+
+class VaultStatus < Sensu::Plugin::Check::CLI
+  option :host,
+         description: 'vault host',
+         short: '-h HOST',
+         long:  '--host HOST',
+         default: '127.0.0.1'
+
+  option :port,
+         description: 'vault port',
+         short: '-p PORT',
+         long: '--port PORT',
+         default: '8200'
+
+  option :scheme,
```

Comment:
I generally dislike using `scheme` for this as in the context of metric gathering scripts it becomes ambiguous if you mean the scheme of the metric path to use or the scheme of the protocol that it needs to use to connect. For this reason I find `protocol` a more accurate and less confusing option name. I will admit that there are a fair number of plugins that have this less than ideal pattern but in those cases it's not worth fixing them but we are trying to avoid these problems with new scripts.

Reply:
Protocol seems pretty loaded too, but I've changed it anyway - it's a better name in this instance

Reply:
Ya I feel you, if you can come up with something better I am all ears.